tools/bin/commands/dev

```diff
@@ -1,15 +1,102 @@
 #!/bin/bash
 
+COOKIE1="$HOME/.syndesis_api_cookie"
+
+
+# helper python command to urlencode
+dev::urlencode() {
+
+local py_script="
+import sys, urllib as ul
+print ul.quote_plus(sys.argv[1])
+"
+python -c "$py_script" $1
+}
+
+# helper python command to unescape html entities
+dev::xml_unescape() {
+
+local py_script="
+import sys, HTMLParser
+parser = HTMLParser.HTMLParser()
+print parser.unescape(sys.argv[1])
+"
+python -c "$py_script" $1
+}
+
 dev::description() {
     echo "Syndesis developer tools"
 }
 
 dev::usage() {
-    cat <<EOT
+    cat <<"EOT"
     --debug <name>            Setup a port forwarding to <name> pod (default: server)
+    --cookie                  Loads a local valid cookie to access Syndesis APIs when using Minishift
+    --refresh                 Used in conjuction with --cookie, forces a refresh of the cookie. 
+                              Ex: curl -k --cookie $(syndesis dev --cookie --refresh)  "https://$(oc get route syndesis  --template={{.spec.host}})/api/v1/connections"
 EOT
 }
 
+extract_and_escape() {
+    local variable=$(grep  -E name=\"$1\" $2 | grep -Po '(?<=value=")[^"]+' )
+    variable="$(dev::xml_unescape $variable)"
+    variable="$(dev::urlencode $variable)"
+    echo "$variable"
+}
+
+dev::create_api_cookie() {
+
+    
+    # second cookie needed because curl flushes old values upon redirect. Might be a bug or some obscure flag missing
+    local cookie2="$(mktemp /tmp/syndesis_cookie.XXX.txt)"
```

Comment:
why do you use an absolute dir for mktemp ? Couldn't we just leave it to the OS where to put the tempdir ? (OS X its a different temp dir, although /tmp is alsow available)